families/supplychain/restapi/openapi.yaml

```diff
@@ -0,0 +1,378 @@
+              data:
+                type: array
+                items:
+                  $ref: "#/definitions/Record"
+              link:
+                $ref: "#/definitions/Link"
+              paging:
+                $ref: "#/definitions/Paging"
+        400:
+          $ref: "#/responses/400BadRequest"
+        500:
+          $ref: "#/responses/500ServerError"
+        503:
+          $ref: "#/responses/503ServiceUnavailable"
+  /records/{record_id}:
+    parameters:
+      - $ref: "#/parameters/record_id"
+    get:
+      summary: Fetches a particular Record.
+      responses:
+        200:
+          description: Successfully fetched Record
+          schema:
+            properties:
+              data:
+                $ref: "#/definitions/Record"
+              link:
+                $ref: "#/definitions/Link"
+        400:
+          $ref: "#/responses/400BadRequest"
+        404:
+          $ref: "#/responses/404NotFound"
+        500:
+          $ref: "#/responses/500ServerError"
+        503:
+          $ref: "#/responses/503ServiceUnavailable"
+
+responses:
+  400BadRequest:
+    description: Request was malformed
+    schema:
+      $ref: "#/definitions/Error"
+  404NotFound:
+    description: Identifier did not match any resource
+    schema:
+      $ref: "#/definitions/Error"
+  500ServerError:
+    description: Something went wrong within the validator
+    schema:
+      $ref: "#/definitions/Error"
+  503ServiceUnavailable:
+    description: API is unable to reach the validator
+    schema:
+      $ref: "#/definitions/Error"
+
+parameters:
+  agent_id:
+    name: agent_id
+    in: path
+    type: string
+    required: true
+    description: Agent identifier, the public key the agent uses.
+  record_id:
+    name: record_id
+    in: path
+    type: string
+    required: true
+    description: "Record identifier, the serial number or natural identifier
+      of the item (material) being tracked."
+  count:
+    name: count
+    in: query
+    type: integer
+    default: 1000
+    description: Number of items to return
+  min:
+    name: min
+    in: query
+    type: string
+    description: Id or index to start paging (inclusive)
+  max:
+    name: max
+    in: query
+    type: string
+    description: Id or index to end paging (inclusive)
+
+definitions:
+  Link:
+    type: string
+    example: https://api.sawtooth.com/agents/65cd3a3ce088b265b626f704b7f3db97b6f12e848dccb35d7806f3d0324c71b709ed360d602b8b658b94695374717e3bdb4b76f77886953777d5d008558247dd
+  Paging:
+    properties:
+      start_index:
+        type: integer
+        example: 1000
+      total_count:
+        type: integer
+        example: 54321
+      previous:
+        type: string
+        example: https://api.sawtooth.com/agents?min=0&count=1000
+      next:
+        type: string
+        example: https://api.sawtooth.com/agents?min=2000&count=1000
+  Error:
+    properties:
+      code:
+        type: integer
+        example: 34
```

Comment:
So, to be clear, I wasn't asking for full error documentation. I think that is premature. I was asking that just this one particular example be a valid error that the API might send back. It's not the end of the world, but I prefer examples to be something you might actually see in the wild whenever possible.

The reason I think this is inherently an _invalid_ example is because `34: Agent Not Found`, defines an error code that already exists in the main REST API and means something else. While technically you could do this, I would strongly object to redefining error codes in any official Sawtooth APIs, because it could cause confusion. I only suggested you use `54: Invalid Paging Query` because it's already defined, and I imagine the SupplyChain API will reuse it.

Anyway. I'm not requesting changes on this. Just wanted to make sure my concern was understood.